src/super_gradients/training/datasets/detection_datasets/yolo_format_detection.py

```diff
@@ -0,0 +1,190 @@
+        if labels_not_in_images:
+            logger.warning(f"{len(labels_not_in_images)} label files are not associated to any image.")
+
+        # Only keep names that are in both the images and the labels
+        valid_base_names = list(unique_image_file_base_names & unique_label_file_base_names)
+        if len(valid_base_names) != len(all_images_file_names):
+            logger.warning(
+                f"As a consequence, "
+                f"{len(valid_base_names)}/{len(all_images_file_names)} images and "
+                f"{len(valid_base_names)}/{len(all_labels_file_names)} label files will be used."
+            )
+
+        self.images_file_names = list(
+            sorted(image_full_name for image_full_name in all_images_file_names if remove_file_extension(image_full_name) in valid_base_names)
+        )
+        self.labels_file_names = list(
+            sorted(label_full_name for label_full_name in all_labels_file_names if remove_file_extension(label_full_name) in valid_base_names)
+        )
+        return len(self.images_file_names)
+
+    def _load_annotation(self, sample_id: int) -> dict:
+        """Load relevant information of a specific image.
+
+        :param sample_id:   Sample_id in the dataset
+        :return:            Dictionary with the following keys:
+            - "target":             Target Bboxes (detection) in XYXY_LABEL format
+            - "initial_img_shape":  Image (height, width)
+            - "resized_img_shape":  Resides image (height, width)
+            - "img_path":           Path to the associated image
+        """
+        image_path = os.path.join(self.images_folder, self.images_file_names[sample_id])
+        label_path = os.path.join(self.labels_folder, self.labels_file_names[sample_id])
+
+        image_width, image_height = imagesize.get(image_path)
+        image_shape = (image_height, image_width)
+
+        yolo_format_target = parse_yolo_label_file(label_path)
+
+        converter = ConcatenatedTensorFormatConverter(input_format=LABEL_NORMALIZED_CXCYWH, output_format=XYXY_LABEL, image_shape=image_shape)
+        target = converter(yolo_format_target)
+
+        # The base class includes a feature to resize the image, so we need to resize the target as well when self.input_dim is set.
+        if self.input_dim is not None:
+            r = min(self.input_dim[0] / image_height, self.input_dim[1] / image_width)
+            target[:, :4] *= r
+            resized_img_shape = (int(image_height * r), int(image_width * r))
+        else:
+            resized_img_shape = image_shape
+
+        annotation = {
+            "target": target,
+            "initial_img_shape": image_shape,
+            "resized_img_shape": resized_img_shape,
+            "img_path": image_path,
+            "id": np.array([sample_id]),
+        }
+        return annotation
+
+
+def parse_yolo_label_file(label_file_path: str) -> np.ndarray:
```

Comment:
why is that not a class method? 
can we make it static and private?

Reply:
Sure
0aec05f9a5d50dceae72974033ea09073817d89b